### ISSUE TITLE
The query configuration parser is too lenient

### ISSUE THREAD

**Issue by Xophmeister** [tweag/topiary#311]
**Describe the bug**
The query configuration parser, which extracts the language and (optionally) indentation offset, uses a regular expression for `(#PREDICATE! VALUE)` tokens in the query language. It matches them anchored to the beginning of a line.

The same syntax can also be used in query rules...

**To Reproduce**
...so write a query rule with a predicate that's at the beginning of a line. For example:

```scm
(some_node
(_) @append_delimiter
(#!delimiter ";")
)
```

It may not be pretty, but this is a valid formatting query, which the configuration parser will think contains a configuration directive and fail with:

```console
[2023-02-17T14:53:55Z ERROR topiary] Unknown configuration predicate 'delimiter'
```

**Expected behavior**
Only root level (i.e., not associated with any query) directives should be parsed as configuration directives.

**Environment**
 - OS name + version: NixOS 22.05
 - Version of the code: 04d7fd8b7368976

**Additional context**
n/a

**Comment by nbacquey** [comment]
The top-level predicates we want to capture in the configuration are stored in the `general_predicates` field of the tree-sitter query, once it has been parsed with `Query::new()`.

The problem, however, is that you need to know the proper grammar to parse a query file, so we have a bootstrapping problem. Maybe some subroutines of `Query::new()` can help?

**Comment by Xophmeister** [comment]
I've added a dependency on https://github.com/nvim-treesitter/tree-sitter-query to parse the query files. I have used that to extract the root-level predicates...I didn't see `Query::general_predicates` :sweat:

Not to worry :slightly_smiling_face: My solution is almost ready for a PR. Just need to write some tests.

**Comment by torhovland** [comment]
Reopened due to #333.

**Comment by torhovland** [comment]
I agree this should be improved, but I would argue that the priority of this issue is less than P1, given that none of our supplied query files, and no reasonably formatted custom query file, are affected.

**Comment by Xophmeister** [comment]
Now that formatting rules exist for our queries, which would change the indentation error and thus stop the regex from matching, I agree that this can be downgraded to P2.

**Comment by Xophmeister** [comment]
If we're planning to use a configuration file, rather than embedding configuration in the query files, should we downgrade the priority of this issue again? (@ErinvanderVeen)

**Comment by ErinvanderVeen** [comment]
With #448, this issue is now no longer valid I believe.
